msrsave/spank_msrsafe.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * The msr-safe Linux kernel module enables user access to read and
+ * write capabilities for a restricted set of whitelisted Model
+ * Specific Registers (MSRs) on x86 platforms.  The purpose of this
+ * slurm plugin is to ensure that MSRs modified within a user’s slurm
+ * job allocation are reset to their original state before the compute
+ * node is returned to the pool available to other users of the
+ * system.  The msr-safe kernel module is targeting HPC systems that
+ * enforce single user occupancy per compute node, and is not
+ * appropriate for systems where compute nodes are shared between
+ * users.  The modifications that one user makes to whitelisted
+ * registers may impact subsequent users of the processor if not
+ * restored.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <signal.h>
+#include <errno.h>
+
+#include "slurm/spank.h"
+
+#define SLURM_SPANK_MSRSAFE_BUFFER_SIZE 1024
+
+SPANK_PLUGIN(msrsafe, 1);
```

Comment:
`SPANK_PLUGIN(msr-safe, 1);`

Reply:
I was really not sure what the character restrictions are for a spank plugin, if '-' is okay I can add this in.

Reply:
I am not sure what the proper formatting is either, but I was looking through https://github.com/grondo/slurm-spank-plugins/ and noticed the spank plugins were either with a hypen or without. I'm not super picky either way, but thought it would make sense to be consistent with the URL: github.com/llnl/msr-safe.

Reply:
Sounds good to me, I'll make the switch.